vnext/Microsoft.ReactNative/ReactHost/ReactInstanceWin.cpp

```diff
@@ -218,23 +218,32 @@ void ReactInstanceWin::Initialize() noexcept {
             std::unique_ptr<facebook::jsi::ScriptStore> scriptStore = nullptr;
             std::unique_ptr<facebook::jsi::PreparedScriptStore> preparedScriptStore = nullptr;
 
+            switch (m_options.LegacySettings.jsiEngine) {
+              case react::uwp::JSIEngine::Hermes:
 #if defined(USE_HERMES)
-            devSettings->jsiRuntimeHolder = std::make_shared<facebook::react::HermesRuntimeHolder>();
-#elif defined(USE_V8)
-            preparedScriptStore =
-                std::make_unique<facebook::react::BasePreparedScriptStoreImpl>(getApplicationLocalFolder());
+                devSettings->jsiRuntimeHolder = std::make_shared<facebook::react::HermesRuntimeHolder>();
+                break;
+#endif
+              case react::uwp::JSIEngine::V8:
+#if defined(USE_V8)
```

Comment:
Should we remove this? If user requests V8 it should not fall back on chakra

Reply:
We do need a sensible default if user says they want to use JSI but the library is not built with V8 support. This just gracefully falls through to the default JSI option today (Chakra)

---
In reply to: [387940527](https://github.com/microsoft/react-native-windows/pull/4234#discussion_r387940527) [](ancestors = 387940527)